File: src/MinCostClimbingStairs.java
public class MinCostClimbingStairs {
    public int minCostClimbingStairs(int[] cost) {
        int currentCost = cost[1], previousCost = cost[0], temp;
        for (int index = 2 ; index < cost.length ; index++) {
            temp = currentCost;
            currentCost = Math.min(currentCost, previousCost) + cost[index];
            previousCost = temp;
        }
        return Math.min(previousCost, currentCost);
    }
}

File: python/min_cost_climbing_stairs.py
from typing import List


class Solution:
    def minCostClimbingStairs(self, cost: List[int]) -> int:
        previousCost, currentCost = cost[0], cost[1]
        index = 2
        while index < len(cost):
            previousCost, currentCost = currentCost, min(previousCost, currentCost) + cost[index]
            index += 1
        return min(previousCost, currentCost)
